src/fe-text/gui-entry.c

```diff
@@ -266,14 +274,19 @@ static void gui_entry_draw_from(GUI_ENTRY_REC *entry, int pos)
 		if (xpos > end_xpos)
 			break;
 
+		if (*col != color) {
```

Comment:
If `col` is set to `NULL` above you get a NPE.

Reply:
I think the intention was that `col` is only set to `NULL` if the loop won't run.

Perhaps this should instead use `entry->colors[i]`, since `col` is equal to `entry->colors + i` at all points during the loop?